doc/proposal/in-progress/07-signing.md

```diff
@@ -0,0 +1,230 @@
+        ]
+      },
+      {
+        "name": "CD-Name"
+      },
+      {
+        "resources": [
+          [
+            {
+              "digest": [
+                {
+                  "hashAlgorithm": "sha256"
+                },
+                {
+                  "normalisationAlgorithm": "manifestDigest/V1"
+                },
+                {
+                  "value": "00000000000000"
+                }
+              ]
+            },
+            {
+              "extraIdentity": [
+                {
+                  "key": "value"
+                }
+              ]
+            },
+            {
+              "name": "Resource1"
+            },
+            {
+              "version": "v0.0.3resource"
+            }
+          ]
+        ]
+      },
+      {
+        "version": "v0.0.1"
+      }
+    ]
+  },
+  {
+    "meta": [
+      {
+        "schemaVersion": "v2"
+      }
+    ]
+  }
+]
+
+```
+
+### Digester
+
+As described, resources have a digest field to store the content hash. Different resource types will use a different normalisationAlgorithm: 
+
+ - `ociArtifactDigest/v1`: uses the hash of the manifest of an oci artifact
+ - `genericBlobDigest/v1`: uses the hash of the blob
+
+## Sign Algorithm
+
+Signing a component-descriptor requires a hash of the normalised component-descriptor, that can be signed.
+The digest fields *MUST* be calculated during normalisation and already existing digest fields *CAN NOT* be trusted. Resources have to be accessed and digested. ComponentReferences has to be followed recursively, calculating the digest for the referenced componentDescriptor. If digests fields for resources or component-references exist, they have to be compared against the calculated digest and rejected if different.
+
+### RSA
+
+After the digest for the normalised component-descriptor is calculated, it can be signed using RSASSA-PKCS1-V1_5 as signature.algorithm. The corresponding signature is stored hex encoded in signature.value with a mediaType of application/vnd.ocm.signature.rsa.
+
+### Signing Server
+
+- Signing Server provides 1 endpoint "/sign" which directly signs the bytes of the request body with RSA
+
+- Component CLI sends Digest of CD to Signing Server and adds the returned signature in PEM format to the CD
+
+## Verification Algorithm
+
+Verifying a component-descriptor consits of three steps. Failing any step **MUST** fail the validation.
+
+1. Verify the digest of all resources and component references. Recursively follow component references and create an in-memory representation of the referenced component-descriptor by accessing and digesting all resources and references. Do not trust any digest data in child component-descriptors. The digest of the normalised in-memory representation of a component-reference **MUST** match the digest in the root component-descriptor (that contains a signature we verify in the next step).
+2. verify the signature, identified by signatureName.
```

Comment:
Wouldn't it be better to first verify the signature and only if this succeeds, continue with the digest validation?

Reply:
In theory, I think the order of the steps does not matter.

Checking the signature first would save computation in case of a manipulated signature. 
I suggest adding a mark that the order of the steps can be changed. Wdyt?

Reply:
You are right, the order does not change the principal behaviour and it improves the performance only in case of a modification. I guess we do not need to change the order.